beamer/health/check.py

```diff
@@ -0,0 +1,526 @@
+        web3 = make_web3(rpc)
+        assert chain_id == ChainId(web3.eth.chain_id)
+
+        info = deployment_info[ChainId(chain_id)]
+        contracts = beamer.agent.contracts.make_contracts(web3, info)
+
+        ef = beamer.agent.events.EventFetcher(
+            web3, (contracts["RequestManager"], contracts["FillManager"]), start_block
+        )
+
+        events[chain_id] = ef.fetch()
+
+    return cast(ChainEventMap, events)
+
+
+def make_web3(rpc: str) -> Web3:
+    web3 = Web3(Web3.HTTPProvider(rpc, request_kwargs=dict(timeout=5)))
+    web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+    return web3
+
+
+def get_agent_liquidity(
+    agent_address: str, rpcs: dict, tokens: TokenMap
+) -> dict[str, dict[ChainId, int]]:
+    contract_abi = load_ERC20_abi()
+    liquidity: dict = defaultdict(dict)
+    agent_address = Web3.toChecksumAddress(agent_address)
+
+    for name, chain_to_token_mapping in tokens.items():
+        for chain_id, token_address in chain_to_token_mapping:
+            contract_address = Web3.toChecksumAddress(token_address)
+            web3 = make_web3(rpcs[int(chain_id)][0])
+            contract = web3.eth.contract(address=contract_address, abi=contract_abi)
+            token_decimals = contract.functions.decimals().call()
+            liquidity[name][chain_id] = (
+                contract.functions.balanceOf(agent_address).call() * 1 * 10**-token_decimals
+            )
+
+    for chain_id, (rpc, _start_block) in rpcs.items():
+        web3 = make_web3(rpc)
+        agent_address = Web3.toChecksumAddress(agent_address)
+        assert ChainId(int(chain_id)) == ChainId(web3.eth.chain_id)
+        liquidity["eth"][chain_id] = web3.eth.get_balance(agent_address) * 1 * 10**-18
+
+    return liquidity
+
+
+def create_transfers_object(events_grouped_by_chain_id: ChainEventMap) -> TransferMap:
+    transfers = defaultdict(dict)  # type: ignore[var-annotated]
+
+    for events in events_grouped_by_chain_id.values():
+        for x in events:
+            if isinstance(x, beamer.agent.events.RequestCreated):
+                transfers[x.request_id.hex()]["created"] = x
+            elif isinstance(x, beamer.agent.events.RequestFilled):
+                transfers[x.request_id.hex()]["filled"] = x
+            elif isinstance(x, beamer.agent.events.DepositWithdrawn):
+                transfers[x.request_id.hex()]["withdrawn"] = x
+            elif isinstance(x, beamer.agent.events.ClaimMade):
+                claims = transfers[x.request_id.hex()].get("claimed", {})
+
+                if x.claim_id not in claims:
+                    claims[x.claim_id] = []
+
+                claims[x.claim_id].append(x)
+
+                transfers[x.request_id.hex()]["claimed"] = claims
+
+    return cast(TransferMap, transfers)
+
+
+def get_block(chain_id: int, block_number: int) -> BlockData:
+    rpc = get_config()["rpcs"][chain_id][0]
+
+    web3 = make_web3(rpc)
```

Comment:
Please consider reusing `Web3` instances as this is making an entirely new connection to the RPC each time you request a block and may be inefficient. I am fine with it if it would be a bigger change, so it isn't critical for this PR, but at least we should have it in mind.

Reply:
Actually, I don't know why I have get_block anymore. It seems that I don't use it anywhere.